cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,325 @@
+    }
+
+    List<StructuredRow> sortedResults = doSort(results.get(), sortRequest);
+    return sortedResults.stream().map(this::fromRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Helper method to apply the sorting onto a list of rows. Sorting needs to take place at the store-level so it can
+   * leverage the StructuredRow to enable sorting on any field.
+   *
+   * @param rows list of {@link StructuredRow} to be sorted
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    if (!fieldTypes.containsKey(sortField)) {
+      throw new IllegalArgumentException(
+        String
+          .format("Invalid value '%s' for sortBy. This field does not exist in the Drafts table.",
+                  sortField));
+    }
+
+    rows.sort((o1, o2) -> {
+      FieldType.Type fieldType = fieldTypes.get(sortField);
+      Comparator<StructuredRow> comparator = null;
+      switch (fieldType) {
+        case STRING:
+          comparator = Comparator.<StructuredRow, String>comparing(o -> o.getString(sortField));
+          break;
+        case INTEGER:
+          comparator = Comparator.<StructuredRow, Integer>comparing(o -> o.getInteger(sortField));
+          break;
+        case LONG:
+          comparator = Comparator.<StructuredRow, Long>comparing(o -> o.getLong(sortField));
+          break;
+        case FLOAT:
+          comparator = Comparator.<StructuredRow, Float>comparing(o -> o.getFloat(sortField));
+          break;
+        case DOUBLE:
+          comparator = Comparator.<StructuredRow, Double>comparing(o -> o.getDouble(sortField));
+          break;
+        case BYTES:
+          comparator = Comparator.comparing(o -> o.getBytes(sortField), Bytes.BYTES_COMPARATOR);
+          break;
+        default:
+          throw new NotImplementedException(String.format("Cannot sort field '%s' because type '%s' is not supported.",
+                                                          sortField, fieldType.toString()));
+      }
+
+      if (sortRequest.getOrder() != SortRequest.SortOrder.ASC) {
+        comparator = comparator.reversed();
+      }
+      return comparator.compare(o1, o2);
+    });
+
+    return rows;
+  }
+
+  /**
+   * Fetch a given draft if it exists
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   * @return an {@link Optional<Draft>} representing the requested draft
+   */
+  public Optional<Draft> getDraft(DraftId id) throws RuntimeException {
+    AtomicReference<Optional<Draft>> result = new AtomicReference<>();
```

Comment:
There is a TransactionRunners utility class that you can use instead of doing this.

Reply:
Ah thats much cleaner, refactored everything to use TransactionRunners